src/loader/load_sonar_dataset.py

```diff
@@ -43,37 +43,42 @@ def load_sonar_dataset(dataset_path: str, limit_n_recs: int = None, multiprocess
 
     # recording
     recording_folder_names = get_subfolder_names(dataset_path)
+
     recording_folder_names = [
         os.path.join(dataset_path, recording_folder_name)
         for recording_folder_name in recording_folder_names
     ]
-
-    if limit_n_recs is not None:
-        recording_folder_names = recording_folder_names[:limit_n_recs]
     
-    recordings = None
+    if limit_n_recs is not None:
+        enumerated_recording_folder_names = list(enumerate(recording_folder_names[:limit_n_recs]))
+    else:
+        enumerated_recording_folder_names = list(enumerate(recording_folder_names))
+
+    # USE ONE (Multiprocessing or Single Thread)
+    # Multiprocessing:
     if multiprocessing:
-        # USE ONE (Multiprocessing or Single Thread)
-        # Multiprocessing:
         pool = Pool()
         recordings = pool.imap_unordered(
-            read_recording_from_folder, recording_folder_names, 10
+            read_recording_from_folder, enumerated_recording_folder_names, 10
         )
         pool.close()
         pool.join()
+
+    # Single thread:
     else:
-        # Single thread:
         recordings = [read_recording_from_folder(recording_folder_name, continue_on_error = True) for recording_folder_name in recording_folder_names]
 
     recordings = list(filter(lambda x: x is not None, recordings))
     assert len(recordings) > 0, "load_sonar_dataset: recordings empty!"
     return recordings
 
 
-def read_recording_from_folder(recording_folder_path: str, continue_on_error: bool = True):
-    try: 
-        subject = get_subject_folder_name(recording_folder_path)
-        return create_recording(recording_folder_path, subject)
+def read_recording_from_folder(enumerated_recording_folder_names: 'tuple(int, str)', continue_on_error: bool = True):
+    recording_folder_path = enumerated_recording_folder_names[1]
+    recording_idx = enumerated_recording_folder_names[0]
```

Comment:
also packst du in Opportunity die Zahlen 0-19 rein? Joa, kann man machen, ich hatte gedacht man baut vielleicht sogar ein string, wo das subject drin encodiert ist, andere Charakteristiken

Reply:
Ja, ich zähle die Recordings nur auf, falls anderes benötigt wird, könnte man sich immernoch ein übersetzendes dict bauen (bzw. recording_idx zu nem string o.Ä. umzubauen, ist auch nicht sonderlich aufwendig, würde ich aber dann demjenigen überlassen, der das in Zukunft braucht)